### PR TITLE
[WIP] feat: refactor Button component to match Fiori 3 specs

### PR DESCRIPTION
**Pull request by rengare**
## Related Issue
Closes SAP/fundamental-styles#417

## Description
Changes nade to the button component according to fiori3 specs

### CODE REVIEW

File: src/_settings.scss [hunk 1]
```diff
@@ -265,8 +265,7 @@ $fd-width--gutter: map-get($fd-spacing, s); // desktop default
 
 // Form Configuration
 $fd-forms-height: $fd-spacing--base * 9 !default; // 36px
-$fd-forms-height--compact: $fd-spacing--base * 7 !default; // 28px
-$fd-forms-height--input-text: $fd-forms-height !default; // deprecate
+$fd-forms-height--compact: $fd-spacing--base * 6.5 !default; // 26px before 28px
```
Comment (JKMarkowski) [inline src/_settings.scss]: This is also used in `fd-form-select` class (native select component). Should it be changed that way? I saw that we have button configuration below (278-281 lines)
image

File: src/button.scss [hunk 2]
```diff
@@ -220,7 +221,7 @@ a.#{$block} {
 
       // placements
       &::before {
-        margin-right: fd-space(2);
+        margin-right: $fd-button-icon-margin-right;
```
Comment (JKMarkowski) [inline src/button.scss]: Could you also apply these changes to RTL mode?

**Comment by InnaAtanasova** [comment]
There are too many tests updated. Since our components are self-contained you shouldn't update tests that are not testing the buttons.

File: src/_settings.scss (hunk 1 above)
Comment (rengare) [reply src/_settings.scss]: I thin so, You can check the message https://github.com/SAP/fundamental-styles/issues/417#issuecomment-550999964

File: src/button.scss (hunk 2 above)
Comment (rengare) [reply src/button.scss]: sure

File: src/_settings.scss (hunk 1 above)
Comment (InnaAtanasova) [reply src/_settings.scss]: I think Jedrzej is right. Changing the form-height affects all forms and hence all the tests that are updated. The best way to "solve" this issue is to create new variables where you can declare the button height in cosy and compact mode

Comment (rengare) [reply src/_settings.scss]: done, the default value was restored and new variable inside the button file introduced

File: src/button.scss (hunk 2 above)
Comment (rengare) [reply src/button.scss]: fixed

**Comment by rengare** [comment]
> There are a couple of things that I noticed that are not matching Fiori3 specs:
> Cosy button
> 
>     * Cosy button has a height of 2.25rem and clickable height of 2.75rem (Compact button has a height of 1.625rem and clickable height of 2rem)
> 
>     * Paddings left and right are 0.625rem (or 10px) including border
> 
>     * Icon and Text button has wrong paddings/margins
> 
>     * should set a min width

**Comment by rengare** [comment]
@InnaAtanasova 
-     Cosy button has a height of 2.25rem and clickable height of 2.75rem (Compact button has a height of 1.625rem and clickable height of 2rem) -> **I can't fix it without breaking changes**
-     Paddings left and right are 0.625rem (or 10px) including border -> **fixed**
-     Icon and Text button has wrong paddings/margins -> **fixed**
-     should set a min width -> **fixed**

**Comment by JKMarkowski** [comment]
image
The compact mode buton has not centered text. Let's change line-height to `1rem`, it should help.

**Comment by InnaAtanasova** [comment]
Let's discuss the Breaking Changes with @droshev after the holidays.

**Comment by stefanoScalzo** [comment]
Check the visual regression tests, I don't believe you need all of them

File: src/button.scss [hunk 3]
```diff
@@ -261,6 +269,20 @@ a.#{$block} {
     }
   }
 
+  &--text-alignment {
+    &-left {
```
Comment (stefanoScalzo) [inline src/button.scss]: We should make left as the default and only use the center or right as modifiers

**Comment by bcullman** [comment]
can we see a before and after screen shots on the PR description?

**Comment by JKMarkowski** [comment]
@rengare  There is still to much test images changed. Overflow is also missing, but it's probably gonna be solved in other PR

**Comment by sKudum** [comment]
> @rengare There is still to much test images changed. Overflow is also missing, but it's probably gonna be solved in other PR

Hi @JKMarkowski , PR#600 is closed as per Deno's comments.

**Comment by rengare** [comment]
closed due to problems with build. New PR was made as continuation #624

**Pull request closed without merging**